homeassistant/components/lyric/__init__.py

```diff
@@ -157,9 +165,59 @@ class LyricDeviceEntity(LyricEntity):
     @property
     def device_info(self) -> DeviceInfo:
         """Return device information about this Honeywell Lyric instance."""
-        return DeviceInfo(
-            connections={(dr.CONNECTION_NETWORK_MAC, self._mac_id)},
-            manufacturer="Honeywell",
-            model=self.device.deviceModel,
-            name=self.device.name,
-        )
+        return _get_lyric_device_info(self.device)
+
+
+def _get_lyric_device_info(device: LyricDevice) -> DeviceInfo:
+    return DeviceInfo(
+        connections={(dr.CONNECTION_NETWORK_MAC, device.macID)},
+        manufacturer="Honeywell",
+        model=device.deviceModel,
+        name=device.name,
+    )
+
+
+class LyricRoomEntity(CoordinatorEntity[DataUpdateCoordinator[Lyric]]):
+    """Defines a Honeywell Lyric room entity."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator[Lyric],
+        location: LyricLocation,
+        device: LyricDevice,
+        room: LyricRoom,
+        key: str,
+    ) -> None:
+        """Initialize the Honeywell Lyric entity."""
+        super().__init__(coordinator)
+        self._key = key
+        self._location = location
+        self._mac_id = device.macID
+        self._room_id = room.id
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID for this entity."""
+        return self._key
+
+    @property
+    def location(self) -> LyricLocation:
+        """Get the Lyric Location."""
+        return self.coordinator.data.locations_dict[self._location.locationID]
+
+    @property
+    def device(self) -> LyricDevice:
+        """Get the Lyric Device."""
+        return self.location.devices_dict[self._mac_id]
+
+    @property
+    def room(self) -> LyricRoom:
+        """Get the Lyric Room."""
+        return self.coordinator.data.rooms_dict[self._mac_id][self._room_id]
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return device information about this Honeywell Lyric instance."""
+        return _get_lyric_device_info(self.device)
```

Comment:
Set this in `__init__` instead:
```py
        self._attr_device_info = _get_lyric_device_info(self.device)

Reply:
Same here: I'd like to keep this as close as possible to the existing `LyricEntity` class unless there's a good reason for it to be different.

I also personally prefer to use properties for data that is derived from a member so that you can't have members that get out of sync.